examples/retrieval/README.md

```diff
@@ -0,0 +1,6 @@
+# llama.cpp/examples/retrieval
+
+Demonstration of simple retrieval technique based on cosin similarity
+
+More info:
+https://github.com/ggerganov/llama.cpp/pull/6193
```

Comment:
For the README, I think it's more useful to at least have an example command to run retrieval, you can copy the description of this PR into this file.